walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -92,6 +92,31 @@ public boolean overlaps(TimeRange other) {
     return this.contains(other.start) || other.contains(this.start);
   }
 
+  /**
+  * Returns the overlapping time window between two TimeRange elements.
+  * If there is no overlap, throws exception.
+  */
+  public TimeRange getOverlap(TimeRange other) {
+    if (!this.overlaps(other)) {
+      throw new IllegalArgumentException("TimeRange arguments do not overlap.");
+    }
+
+    //Should I use .start() or .start here?
```

Comment:
Both are okay. Maybe use start() to be consistent.

Reply:
Good point, thank you.
233ab83